docs/pipelines/process/run-number.md

```diff
@@ -99,6 +99,14 @@ $(Build.DefinitionName)_$(Build.DefinitionVersion)_$(Build.RequestedFor)_$(Build
 
 The first four variables are predefined. `My.Variable` is defined by you on the [variables tab](variables.md).
 
+### Expressions
+
+You can also use [Expressions](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/expressions?view=azure-devops) to dynamically change the build version number depending on certain conditions. For example, if the build is triggered from a pull request, then we can use that in the number format, otherwise fall back to the builds source branch name, or even date format if this variable is also not available:
+
+```yaml
+name: ${{ coalesce( '$(System.PullRequest.PullRequestNumber)', '$(System.PullRequest.SourceBranch)', '$(Date:yyyyMMdd)') }}$(Rev:.r)
```

Comment:
Seems as though `$(System.PullRequest.PullRequestNumber)` doesn't collapse into a null string. This example may not work correctly.

https://github.com/microsoft/azure-pipelines-yaml/issues/474

Reply:
I was eventually able to get this to work using:
```
variables:
  ${{ if eq( variables['build.reason'], 'PullRequest' ) }}:
    date: ''
  ${{ if ne( variables['build.reason'], 'PullRequest' ) }}:
    date: ${{ '$(Date:yyyyMMdd)' }}

name: ${{ coalesce( variables['date'], '$(System.PullRequest.PullRequestNumber)', '$(Date:yyyyMMdd)' ) }}$(Rev:.r)
```